opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -439,11 +516,8 @@ def set_tracer_provider(tracer_provider: TracerProvider) -> None:
     """
     global _TRACER_PROVIDER  # pylint: disable=global-statement
 
-    if _TRACER_PROVIDER is not None:
-        logger.warning("Overriding of current TracerProvider is not allowed")
-        return
```

Comment:
I don't think we would want to allow overriding the global tracer provider once set.

Reply:
Both options have pros and cons. I have no preference

Reply:
I believe the exporters will rely one global tracer provider to fetch the `Resource` and convert the attributes to tags when exporting. As if now each span created has the resource attribute and it remains same for all the spans from tracer provider instance and spec doesn't have resource attribute for span. This is one such place I can think of but other can provide more if any.

Reply:
Done